command_line/slice_sequence.py

```diff
@@ -158,33 +160,42 @@ def run(self, args=None):
 
         # check if slicing into blocks
         if params.block_size is not None:
-            if slice_exps:
-                if len(experiments) > 1:
-                    raise Sorry(
-                        "For slicing into blocks please provide a single " "scan only"
-                    )
-                scan = experiments[0].scan
+            if not slice_exps:
+                raise Sorry(
+                    "For slicing into blocks, an experiment file must be provided"
+                )
+
+            if len(experiments) > 1:
+                raise Sorry(
+                    "For slicing into blocks please provide a single " "scan only"
+                )
+            scan = experiments[0].scan
 
             # Having extracted the scan, calculate the blocks
             params.image_range = calculate_block_ranges(scan, params.block_size)
 
             # Do the slicing then recombine
-            if slice_exps:
-                sliced = [
-                    slice_experiments(experiments, [sr])[0] for sr in params.image_range
-                ]
-                sliced_experiments = ExperimentList()
-                for exp in sliced:
-                    sliced_experiments.append(exp)
+            sliced = [
+                slice_experiments(experiments, [sr])[0] for sr in params.image_range
+            ]
+            sliced_experiments = ExperimentList()
+            generate_experiment_identifiers(sliced)
+            for exp in sliced:
+                sliced_experiments.append(exp)
```

Comment:
I think this should work and be simpler?
```suggestion
            generate_experiment_identifiers(sliced)
            sliced_experiments = ExperimentList(sliced)
```